code/game/machinery/doors/airlock.dm

```diff
@@ -1016,6 +1044,24 @@ About the new airlock wires panel:
 	if(!forced)
 		if( !hasPower() || isWireCut(AIRLOCK_WIRE_OPEN_DOOR) )
 			return 0
+	if(sabotaged && sabotageTurf)
+		p_open = 1
+		update_icon()
+		visible_message("<span class='warning'>[src]'s panel is blown off in a spray of deadly shrapnel!</span>")
+		explosion(sabotageTurf,-1,1,1)

```

Comment:
I wanted to make the door charge like the requester (Bluespace)'s specifications. He wanted it to only explode on the turf it was applied from and not to damage the airlock (besides the panel). That's why some stuff might seem strange.

As for the drone dispenser, you're right and I'll change the clamp. However, I didn't have an issue with the stack not deleting.